### ISSUE TITLE
Alternative way to use the pre-trained sentence embedding models

### ISSUE THREAD

**Issue by schrum2** [schrum2/MarioDiffusion#68]
Currently, we take the whole sentence and send it to the sentence transformer to get a single embedding vector, which is then unsqueezed to add an extra dimension that makes the shape compatible with what we are doing with my Transformer model. However, the training data is actually a collection of phrases separated by periods, a bunch of individual sentences. It occurs to me that we should have an option (maintain backwards compatibility) to take those captions, split them into several sentences, and then concatenate them along that dimension that is currently always 1 in the current implementation. Embedding the individual caption phrases could allow for much more precise learning conditioned on the text.

This should work. The one challenging issue I anticipate is that if a batch of captions is sent, and different ones have a different number of period-separated phrases, there will be an issue with the different samples being of different lengths. However, we already have a solution for this in the token-embedding case: just pad the shorter inputs with special padding vectors at the end, which will just be the embedding of the empty string.

We need to give the pre-trained models a fair shake, and I think this could improve their performance.
